sdk/identity/identity/samples/AzureCacheForRedis/node-redis.md

```diff
@@ -0,0 +1,278 @@
+- [Authenticate with Azure AD - Using Token Cache](#authenticate-with-azure-ad-using-token-cache)
+- [Troubleshooting](#troubleshooting)
+
+#### Prerequisites
+
+- Configuration of Role and Role Assignments is required before using the sample code in this document.
+- **Dependency Requirements:**
+    Add the following dependencies to *package.json*:
+
+    ```
+    "dependencies": {
+      "@azure/identity": "^2.0.4",
+      "redis": "^4.1.0"
+      }
+    ```
+- Familiarity with the [node-redis](https://github.com/redis/node-redis) and [Azure Identity for JavaScript](https://docs.microsoft.com/javascript/api/overview/azure/identity-readme?view=azure-node-latest) client libraries is assumed.
+
+#### Samples Guidance
+
+* [Authenticate with Azure AD - Hello World](#authenticate-with-azure-ad-hello-world):
+   This sample is recommended for users getting started to use Azure AD authentication with Azure Cache for Redis.
+* [Authenticate with Azure AD - Handle Reauthentication](#authenticate-with-azure-ad-handle-reauthentication):
+   This sample is recommended to users looking to build long-running applications and would like to handle reauthenticating with Azure AD upon token expiry.
+* [Authenticate with Azure AD - Using Token Cache](#authenticate-with-azure-ad-using-token-cache):
+  This sample is recommended to users looking to build long-running applications that would like to handle reauthenticating with a token cache. The token cache stores and proactively refreshes the Azure AD access token 2 minutes before expiry and ensures a non-expired token is available for use when the cache is accessed.
+
+#### Authenticate with Azure AD: Hello World
+
+This sample is intended to assist in authenticating with Azure AD via the node-redis client library. It focuses on displaying the logic required to fetch an Azure AD access token and to use it as password when setting up the node-redis instance.
+
+##### Migration guidance
+
+When migrating your existing application code, replace the password input with the Azure AD token. Azure Redis Cache name, Service Principal Username, Azure AD token, and use of SSL are required while connecting with the cache. The username will depend on whether you are using Service Principal, Managed identity, or Microsoft username.
+
+Integrate the logic in your application code to fetch an Azure AD access token via the Azure Identity library, as shown below. Replace it with the password configuring/retrieving logic in your application code.
+
+```ts
+import { createClient } from "redis";
+import { DefaultAzureCredential } from "@azure/identity";
+import * as dotenv from "dotenv";
+dotenv.config();
+
+async function main() {
+  // Construct a Token Credential from Identity library, e.g. ClientSecretCredential / ClientCertificateCredential / ManagedIdentityCredential, etc.
+  const credential = new DefaultAzureCredential();
+
+  // The scope will be changed for Azure AD Public Preview
```

Comment:
If you haven't already, please create an issue to track updating this scope value.

Reply:
https://github.com/Azure/azure-sdk-for-js/issues/22770